src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -628,7 +628,6 @@ this session.`;
             if (!recipientSubmissionFormModel.isValid) {
               failToSaveQuestions[questionSubmissionFormModel.questionNumber] =
                   'Invalid responses provided. Please check question constraints.';
-              return;
```

Comment:
Upon updating `isValid` in `constsum-recipients-question-constraint.component.ts`, the error message returned was no longer correct:
<img width="852" alt="Screen Shot 2022-01-16 at 5 31 54 PM" src="https://user-images.githubusercontent.com/54243224/149656751-4d90d098-57fd-4b76-8bb3-11399a381261.png">
The issue stems from this return statement, as it causes the adding of the response to the `responses` array (lines 638-644) to be skipped. Removing this allows the response to be sent to the backend for proper error handling.

Reply:
Not sure if I can reproduce this -- changing `isValid` there should affect only the constsum recipients question, and testing it with question 8 and with the `return` statement in place, the expected error message shows up for me:
<img width="299" alt="Screenshot 2022-01-20 at 10 55 48 AM" src="https://user-images.githubusercontent.com/41856541/150263943-177ea1ca-8813-4d32-aa50-f0ae2594f539.png">

<img width="683" alt="Screenshot 2022-01-20 at 10 48 05 AM" src="https://user-images.githubusercontent.com/41856541/150263144-e34e6969-e206-4f99-855c-90030fdd107f.png">

Am I missing something here? My understanding is that after setting the error message to be "Invalid responses provided...", adding the return statement terminates it before backend processing, which works as intended (raise a error in frontend without needing to send a request to backend)

Reply:
From what I understand, it seems that while the return statement prevents the responses from being pushed into the `responses` array, it does not prevent the `feedbackResponseRequest` from being pushed into the `savingRequests` array (line 647 onwards). Therefore, an empty question response is sent to the backend, which causes the wrong error message is returned.

Given this, should I add a check that prevents requests from being sent to the backend if the question is invalid?

Reply:
It looks like the `return` statement was intended to terminate the entire question form submission and break out before anything is pushed into `savingRequests`, but you're right that it happens nonetheless -- my guess is that it's probably due to the `return` being in `forEach`, which doesn't guarantee breaking out of the function? [(link)](https://stackoverflow.com/questions/34653612/what-does-return-keyword-mean-inside-foreach-function)

I wasn't able to reproduce your behaviour locally, but yes on staging it happened for me about 50% of the time!

Either way, that seems to affects the main branch as well, so we might want to open another issue for it. For this one at least we can probably leave the session submission logic as it is?

Reply:
Got it! I have reverted this change, and created a new issue #11539 to address this.